scalanet/src/io/iohk/scalanet/peergroup/InMemoryPeerGroup.scala

```diff
@@ -5,22 +5,24 @@ import monix.reactive.{MulticastStrategy, Observable}
 import monix.reactive.subjects.ConcurrentSubject
 import monix.execution.Scheduler.Implicits.global
 import InMemoryPeerGroup._
+import io.iohk.scalanet.peergroup.PeerGroup.ServerEvent
+import io.iohk.scalanet.peergroup.PeerGroup.ServerEvent.ChannelCreated
 
 import scala.collection.concurrent.TrieMap
-import scala.util.Random
 
 class InMemoryPeerGroup[A, M](address: A)(implicit network: Network[A, M]) extends PeerGroup[A, M] {
 
-  private var status: PeerStatus = PeerStatus.NotInitialized
-  private val channelStream = ConcurrentSubject[InMemoryChannel[A, M]](MulticastStrategy.publish)
-  private val channelsMap: TrieMap[ChannelID, InMemoryChannel[A, M]] = TrieMap()
-  private def receiveMessage(channelID: ChannelID, from: A, msg: M): Unit = {
+  private[peergroup] var status: PeerStatus = PeerStatus.NotInitialized
+  private[peergroup] val channelStream = ConcurrentSubject[ServerEvent[A, M]](MulticastStrategy.publish)
+  private[peergroup] val channelsMap: TrieMap[ChannelID, InMemoryChannel[A, M]] = TrieMap()
+
+  def receiveMessage(channelID: ChannelID, from: A, msg: M): Unit = {
```

Comment:
I am not following, why couldn't the above members (`status`, `channelStream` and `channelMap`) be left as private and this particular method neither? All of those should be used only by the internal implementation and seems to not be used by any other entity in this PR.

Reply:
I moved the creation of test instances into the test directory which then requires access to the internal state. We can discuss PeerUtils et al a bit later.

Reply:
I see, I am fine to leave the proper solution for the test pack PR, but I will require for these methods to be private again in the final solution.